lib/Sema/CSDiagnostics.cpp

```diff
@@ -2677,6 +2677,12 @@ void ContextualFailure::tryFixIts(InFlightDiagnostic &diagnostic) const {
 }
 
 bool ContextualFailure::diagnoseMissingFunctionCall() const {
+  // Don't suggest inserting a function call if the function expression
+  // isn't written explicitly in the source code.
+  auto *anchor = getAsExpr(simplifyLocatorToAnchor(getLocator()));
+  if (!anchor || anchor->isImplicit())
+    return false;
```

Comment:
It's annoying that there's a bit of duplicated code here. I wonder if we should always try to diagnose this through `InsertExplicitCall`?

Reply:
Yes, that would be great. I think that this is just a leftover from the days of CSDiag, most likely not all the locations have been refactored to produce a tailored `InsertExplicitCall` instead of `IgnoreContextualMismatch` but it was shadowed by the fact that they produce the same diagnostic. Maybe you should just try deleting this whole method and see whether it would be possible to produce a tailored fix where appropriate?

Reply:
✅